File: 495-teemo-attacking/leochang/495.cpp
class Solution {
public:
    int findPoisonedDuration(vector<int>& timeSeries, int duration) {
        int ans = 0;
        if (timeSeries.size() == 0) return 0;
        int end = timeSeries[0];
        for (auto time : timeSeries) {
            if (time < end) {
                ans -= (end - time);
            }
            end = time + duration;
            ans += duration;
        }
        return ans;
    }
};

File: 495-teemo-attacking/ken/495.py
class Solution(object):
    def findPoisonedDuration(self, timeSeries, duration):
        """
        :type timeSeries: List[int]
        :type duration: int
        :rtype: int
        """

        if len(timeSeries) == 0:
            return 0

        ans = 0
        start = timeSeries[0]
        for t in timeSeries:
            ans += min(t - start, duration)
            start = t

        ans += duration

        return ans
